src/modules/grubcfg/main.py

```diff
@@ -35,13 +34,10 @@ def modify_grub_default(partitions, root_mount_point, distributor):
     default_dir = os.path.join(root_mount_point, "etc/default")
     default_grub = os.path.join(default_dir, "grub")
     distributor_replace = distributor.replace("'", "'\\''")
-    plymouth_bin = libcalamares.utils.target_env_call(["sh", "-c", "which plymouth"])
     use_splash = ""
     swap_uuid = ""
 
-    libcalamares.utils.debug("which plymouth exit code: {!s}".format(plymouth_bin))
-
-    if plymouth_bin == 0:
+    if libcalamares.globalstorage.value("hasPlymouth"):
```

Comment:
Best if you check if `libcalamares.globalstorage.contains("hasPlymouth")` before getting its value, in case the distro isn't using the plymouthcfg module.